studio/schemas/documents/languageDetails.ts

```diff
@@ -0,0 +1,71 @@
+import { defineType } from "sanity";
+import { languages } from "languages";
+
+const languageOptions = languages.map((language) => {
+  return { title: language.title, value: language.id };
+});
+
+export const languageID = "languagesDetails";
+
+const languageDetails = defineType({
+  name: languageID,
+  type: "document",
+  fields: [
+    {
+      name: "languages",
+      type: "array",
+      description:
+        "Select languages available for translation and designate one as the default language for the homepage.",
+      of: [
+        {
+          type: "object",
+          fields: [
+            {
+              name: "language",
+              type: "string",
+              description: "Select the language available for translation",
+              title: "Language",
+              options: {
+                list: languageOptions,
+              },
+            },
+            {
```

Comment:
We can merge this PR, but I think a better approach would be to have an array of languages with language value, instead of language object + bool.

And then add a new field for setting the default language:

<img width="1425" alt="Screenshot 2024-09-16 at 12 00 45" src="https://github.com/user-attachments/assets/56b0a340-53ba-4cd2-ac44-7072fce240df">

However we're in need of a custom component in that case.

Reply:
Agree with this!